pkg/plugins/platform/router.go

```diff
@@ -84,3 +85,56 @@ func StartDockerRouter() {
 		log.Fatalf("Could not start router: %v", err)
 	}
 }
+
+// determineRouterPorts returns a list of port mappings retrieved from running site
+// containers defining VIRTUAL_PORT env var
+func determineRouterPorts() []string {
+	var routerPorts []string
+	containers, err := util.GetDockerContainers(false)
+	if err != nil {
+		log.Fatal("failed to retreive containers for determining port mappings", err)
+	}
+
+	// loop through all containers with site-name label
+	for _, container := range containers {
+		if _, ok := container.Labels["com.ddev.site-name"]; ok {
+			var exposePorts []string
+
+			httpPorts := util.GetContainerEnv("HTTP_EXPOSE", container)
```

Comment:
Absolutely not recommending we do this right now, but just pointing out that the code here would definitely be simplified by moving to labels for these values.